cli/helpers/sca_code_scanner.py

```diff
@@ -1,17 +1,59 @@
 import click
 from typing import List, Optional
+from git import Repo, GitCommandError
 from cli.utils.shell_executor import shell
 from cli.models import Document
 from cli.utils.path_utils import get_file_dir, join_paths
 from cyclient import logger
+from cli.helpers import sca_utils
 
 BUILD_GRADLE_FILE_NAME = 'build.gradle'
 BUILD_GRADLE_KTS_FILE_NAME = 'build.gradle.kts'
 BUILD_GRADLE_DEP_TREE_FILE_NAME = 'gradle-dependencies-generated.txt'
 BUILD_GRADLE_DEP_TREE_TIMEOUT = 180
 
 
-def run_pre_scan_actions(context: click.Context, documents_to_scan: List[Document], is_git_diff: bool = False):
+def perform_pre_commit_range_scan_actions(path: str, from_commit_documents: List[Document],
+                                          from_commit_rev: str, to_commit_documents: List[Document],
+                                          to_commit_rev: str) -> None:
+    repo = Repo(path)
+    add_ecosystem_related_files_if_exists(from_commit_documents, repo, from_commit_rev)
+    add_ecosystem_related_files_if_exists(to_commit_documents, repo, to_commit_rev)
+
+
+def add_ecosystem_related_files_if_exists(documents: List[Document], repo: Repo, commit_rev: str):
+    documents_to_add: List[Document] = []
+    for doc in documents:
+        ecosystem = get_project_file_ecosystem(doc)
+        for ecosystem_project_file in sca_utils.ecosystem_project_files.get(ecosystem):
```

Comment:
check if ecosystem not null
then move the code below to function
`get_ecosystem_project_files` -> return list of file to add and then do extend